### PR TITLE
[5.4] Return json if sucessfull login expects it

### PR DESCRIPTION
**Pull request by Quezler**

### CODE REVIEW

**Comment by Quezler** [comment]
Sibling of laravel/laravel#4269

**Comment by laurencei** [comment]
Probably needs to go to 5.5 branch?

Because it might change behavior on current applications that dont expect it?

File: src/Illuminate/Foundation/Auth/AuthenticatesUsers.php [hunk 1]
```diff
@@ -99,6 +99,10 @@ protected function sendLoginResponse(Request $request)
         $request->session()->regenerate();
 
         $this->clearLoginAttempts($request);
+        
+        if ($request->expectsJson()) {
+            return response()->json([$this->username() => trans('auth.success')], 200);
```
Comment (lucasmichot) [inline src/Illuminate/Foundation/Auth/AuthenticatesUsers.php]: `200` status code is already the default, no need to set it.

Comment (Quezler) [inline src/Illuminate/Foundation/Auth/AuthenticatesUsers.php]: Well this file and files closely related to authentication both have (non 200) status codes, so i personally prefer if ether all have codes, or none do.

**Comment by Quezler** [comment]
@laurencei good point, this and the subling commit might be better of on 5.5, but the members can change the target branch anyways 😄

**Comment by taylorotwell** [comment]
We're not changing this on 5.4.

**Pull request closed without merging**